### PR TITLE
IDEX-4060: add a custom implementation for VirtualFile

### PR DESCRIPTION
**Pull request by svor**
@vparfonov @dimasnurenko @evidolob @azatsarynnyy

### CODE REVIEW

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/tree/VirtualFileInfo.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+     * @return builder for {@link VirtualFileInfo}
+     */
+    public static Builder newBuilder() {
+        return new VirtualFileInfo().new Builder();
+    }
+
+    /**
+     * Builder utility class for {@link VirtualFileInfo}
+     */
+    public class Builder {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/tree/VirtualFileInfo.java]: I would say strange implementation of **Builder**.
Typical builder should be static class that constructs new instance of **VirtualFileInfo** in **build()** method.

Comment (svor) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/tree/VirtualFileInfo.java]: Actually it isn't a classical **Builder** which has **static** keyword. In my case I create the new instance of a non-static inner class using `new VirtualFileInfo().new Builder()`.

File: ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/debug/BreakpointManagerImpl.java [hunk 2]
```diff
@@ -495,19 +496,23 @@ private void onDebugInitialized(Debugger debugger) {
...
-
-                                                                       @Override
-                                                                       public void setProjectConfig(
-                                                                               ProjectConfigDto projectConfig) {
-                                                                       }
-                                                                   });
+            VirtualFileInfo virtualFileInfo = VirtualFileInfo.newBuilder()
+                                                             .setPath(dto.getPath())
+                                                             .setMediaType(dto.getFileMediaType())
+                                                             .setProject(new HasProjectConfig() {
```
Comment (vzhukovskii) [inline ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/debug/BreakpointManagerImpl.java]: Create default implementation of HasProjectConfig inside iterface.

Comment (svor) [inline ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/debug/BreakpointManagerImpl.java]: Improved

**Comment by vparfonov** [comment]
ok

**Pull request merged**